lgc/patch/PatchBufferOp.cpp

```diff
@@ -115,15 +115,8 @@ bool PatchBufferOp::runImpl(Function &function, PipelineState *pipelineState,
 
   const bool changed = (!m_replacementMap.empty());
 
-  for (auto &replaceMap : m_replacementMap) {
-    Instruction *const inst = dyn_cast<Instruction>(replaceMap.first);
-
-    if (!inst)
-      continue;
-
-    if (!isa<StoreInst>(inst))
-      inst->replaceAllUsesWith(UndefValue::get(inst->getType()));
-
+  for (auto &[inst, replacement] : m_replacementMap) {
+    inst->replaceAllUsesWith(UndefValue::get(inst->getType()));
```

Comment:
How come the check that this isn't a store is no longer required (can this case never happen?)

Reply:
I don't think it was ever required. The type of a StoreInst is void and it has no uses, but it is OK (harmless) to call replaceAllUsesWith on it.